odoo/addons/base/tests/test_misc.py

```diff
@@ -507,3 +507,71 @@ def test_readonly_dict(self):
             d.update({'baz': 'xyz'})
         with self.assertRaises(TypeError):
             dict.update(d, {'baz': 'xyz'})
+
+class TestFormatLang(TransactionCase):
+    # /!\ The monetary argument does not need to be tested on version <= 17.1 because it has no effect.
+    def test_value_and_digits(self):
+        self.assertEqual(misc.formatLang(self.env, 100.23, digits=1), '100.2')
+        self.assertEqual(misc.formatLang(self.env, 100.23, digits=3), '100.230')
+
+        # If value is an empty string, it should return an empty string (not 0).
+        self.assertEqual(misc.formatLang(self.env, ''), '')
+
+        # If digits is 'None' (default value), it should default to 2.
+        self.assertEqual(misc.formatLang(self.env, 100), '100.00')
+
+        # Default rounding is 'HALF_EVEN'
```

Comment:
If we put them as the error message to throw
```suggestion
        # Default rounding should be 'HALF_EVEN'
```

Reply:
For this, I think it will be weird on the error message because they **both** need to be True for it to be `HALF_EVEN`. Otherwise, one is correct for `ROUND_HALF_UP` and the other for `ROUND_HALF_DOWN`.